### PR TITLE
Feature: Upgrade.

### PR DESCRIPTION
**Pull request by MartynasZilinskas**

### CODE REVIEW

File: src/abstractions/handler-base.ts [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+    /**
+     * Initial open state value.
+     * By default it gets initial value from props: defaultOpen and open.
+     */
+    protected getInitialOpenValue(): boolean {
+        let open = false;
+
+        if (this.props.defaultOpen != null) {
+            open = this.props.defaultOpen as boolean;
```
Comment (DeividasBakanas) [inline src/abstractions/handler-base.ts]: Maybe use `Boolean(this.props.defaultOpen)`.

File: src/abstractions/section-base.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+> extends React.Component<TProps, TState> {
+    /**
+     * @throws
+     */
+    constructor(props: TProps, context: SectionBaseContext) {
+        super(props);
+
+        if (context.dropdownOnSectionClickCallback == null || context.dropdownOpen == null) {
+            throw new Error(
+                `simplr-dropdown: (BaseHeader) ${(this.constructor as any).name}` + ` must be inside DropdownHandler component.`
```
Comment (DeividasBakanas) [inline src/abstractions/section-base.ts]: Maybe `any` can be removed?

File: src/abstractions/handler-base.ts [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+
+export abstract class HandlerBase<
+    TProps extends HandlerBaseProps = HandlerBaseProps,
+    TState extends HandlerBaseState = HandlerBaseState
+> extends React.Component<TProps, TState> {
+    constructor(props: TProps) {
+        super(props);
+        if (CAN_I_USE_WINDOW_LISTENERS) {
+            window.addEventListener("click", this.onOutsideClick.bind(this));
+            // TODO: Move keyup to listener to handler container
```
Comment (DeividasBakanas) [inline src/abstractions/handler-base.ts]: New or old TODO?

File: src/abstractions/handler-base.ts [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+}
+
+export interface HandlerBaseState {
+    open: boolean;
+}
+
+export interface BaseHandlerChildContext {
+    dropdownOpen: boolean;
+    dropdownOnHeaderClickCallback: Function;
+    dropdownOnSectionClickCallback: Function;
```
Comment (DeividasBakanas) [inline src/abstractions/handler-base.ts]: Can function be specified?

File: src/abstractions/handler-base.ts [hunk 5]
```diff
@@ -0,0 +1,216 @@
+import * as React from "react";
+import * as PropTypes from "prop-types";
+
+import { EventSource } from "../contracts";
+import { CAN_I_USE_WINDOW_LISTENERS, ESCAPE_KEYCODE } from "../helpers";
+
+export interface HandlerBaseProps {
+    defaultOpen?: boolean;
+    open?: boolean;
+    onOpen?: (source: EventSource) => void;
```
Comment (DeividasBakanas) [inline src/abstractions/handler-base.ts]: Maybe move type to contracts?

File: src/abstractions/handler-base.ts (hunk 3 above)
Comment (MartynasZilinskas) [reply src/abstractions/handler-base.ts]: Old one. Not sure if I should leave that.

**Pull request merged**